src/com/university_project/Service.java

```diff
@@ -0,0 +1,222 @@
+
+            Subject philosophyOfLusineMeliqyan = new Subject("Philosophy");
+            philosophyOfLusineMeliqyan.setGrade(8);
+            Subject[] subjectsOfLusineMeliqyan = new Subject[]{philosophyOfLusineMeliqyan};
+            philosophyStudent2.setSubjects(subjectsOfLusineMeliqyan);
+
+            Subject mathOfYanaGharibyan = new Subject("Math");
+            mathOfYanaGharibyan.setGrade(8);
+            Subject philosophyOfYanaGharibyan = new Subject("Philosophy");
+            philosophyOfYanaGharibyan.setGrade(8);
+            Subject[] subjectsOfYanaGharibyan = new Subject[]{mathOfYanaGharibyan, philosophyOfYanaGharibyan};
+            philosophyStudent3.setSubjects(subjectsOfYanaGharibyan);
+
+            Subject philosophyOfGevorgMeliqjanyan = new Subject("Philosophy");
+            philosophyOfGevorgMeliqjanyan.setGrade(8);
+            Subject[] subjectsOfGevorgMeliqjanyan = new Subject[]{philosophyOfGevorgMeliqjanyan};
+            philosophyStudent4.setSubjects(subjectsOfGevorgMeliqjanyan);
+
+            this.allStudentsCount = physicsFacultyFirstGradeStudents.length
+                    + physicsFacultySecondGradeStudents.length
+                    + philosophyFacultyFirstGradeStudents.length
+                    + philosophyFacultySecondGradeStudents.length;
+            this.allStudents = new Student[this.allStudentsCount];
+            int pos = 0;
+            for (Student element : physicsFacultyFirstGradeStudents) {
+                allStudents[pos] = element;
+                pos++;
+            }
+            for (Student element : physicsFacultySecondGradeStudents) {
+                allStudents[pos] = element;
+                pos++;
+            }
+            for (Student element : philosophyFacultyFirstGradeStudents) {
+                allStudents[pos] = element;
+                pos++;
+            }
+            for (Student element : philosophyFacultySecondGradeStudents) {
+                allStudents[pos] = element;
+                pos++;
+            }
+
+        } catch (RuntimeException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void uniFromOneSubjectGPA() {
+        for (int i = 0; i < allStudents.length; i++) {
+            for (int j = 0; j < allStudents[i].getSubjects().length; j++) {
+                if (allStudents[i].getUniversityName().equals("RAU")
+                        && allStudents[i].
+                        getSubjects()[j].getSubjectName().equals("Math")) {
+                    inUniOneSubjectAllGrade += allStudents[i].getSubjects()[j].getGrade();
+                    inUniOneSubjectStudentsCount++;
+                }
+            }
+        }
+
+        this.inUniOneSubjectAverage = inUniOneSubjectAllGrade / inUniOneSubjectStudentsCount;
+        System.out.println("University's average grade from Math is " + this.inUniOneSubjectAverage);
+    }
+
+    public void facultyFromOneSubjectGPA() {
```

Comment:
Here your method must accept university and subject parameters. It is not good when you immediately compare values. For example, you have done that thing with "Physics" Do the same for other methods.